neo/Trie/MPT/MPTNode.cs

```diff
@@ -0,0 +1,90 @@
+using Neo.Cryptography;
+using Neo.IO.Json;
+using System;
+using System.IO;
+using System.Text;
+
+namespace Neo.Trie.MPT
+{
+    public abstract class MPTNode
+    {
+        private UInt256 hash;
+        public bool Dirty { get; private set; }
+        protected NodeType nType;
+
+        protected virtual UInt256 GenHash()
+        {
+            return new UInt256(Crypto.Default.Hash256(this.Encode()));
+        }
+
+        public virtual UInt256 GetHash()
+        {
+            if (!Dirty && !(hash is null)) return hash;
+            hash = GenHash();
+            Dirty = false;
+            return hash;
+        }
+
+        public void SetDirty()
+        {
+            Dirty = true;
+        }
+
+        public MPTNode()
+        {
+            Dirty = true;
+        }
+
+        public byte[] Encode()
+        {
+            using (BinaryWriter writer = new BinaryWriter(new MemoryStream(), Encoding.UTF8, true))
```

Comment:
LeaveOpen must be false
```suggestion
            using (BinaryWriter writer = new BinaryWriter(new MemoryStream(), Encoding.UTF8, false))
```